corehq/apps/userreports/README.md

```diff
@@ -385,6 +385,76 @@ Variables can be any valid numbers (Python datatypes `int`, `float`, and `long`
 
 More examples can be found on practical [examples page](corehq/apps/userreports/examples/examples.md#evaluator-examples).
 
+#### Case --> Form Lookup
+
+```json
+{
+   "type": "get_case_forms",
+   "case_id_expression": {},  # required
+   "filter": {},  # optional - possibly to remove for now in favor of filter expression
+}
+```
+
+Gets a list of forms from a case, optionally filtering (TBD if that belongs here).
+
+#### Filter Expression
+
+```json
+{
+  "type": "filter",  # not sure on name
+  "items_expression": {},  # expression should return a list
+  "filter_expression": {}  # will be called on each item, should return true or false indicating whether to include
+}
+```
+
+This should work just like python's `filter` function.
+
+#### Map Expression
+
+```json
+{
+  "type": "map",  # not sure on name
+  "items_expression": {},  # expression should return a list
+  "map_expression": {}  # expression to be evaluated on each item in the list
+}
+```
+
+In python the equivalent would be `map(lambda x: x.y, my_list)`, which would return a new list with the lambda function applied.
+
+#### Reduce Expression
+
+```json
+{
+  "type": "reduce",  # not sure on name
+  "items_expression":  {},  # expression should return a list
+  "aggregation_fn": "sum"  # one of a fixed set of allowed aggregation functions
```

Comment:
i feel a little guilty suggesting this but what if we just added a sort expression? then we can explicitly sort the docs by whatever property in the UCR we want and the assumption of ordering is no longer required. it's another major extension but i think in the same realm of the other stuff that this PR is adding